completor/prepare_outputs.py

```diff
@@ -91,42 +91,121 @@ def dataframe_tostring(
-                Headers.GAS_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
-                Headers.WATER_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
-                Headers.MAX_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
-                Headers.DEFAULTS: "{:.10s}".format,
-                Headers.WATER_HOLDUP_FRACTION_LOW_CUTOFF: "{:.10g}".format,
-                Headers.WATER_HOLDUP_FRACTION_HIGH_CUTOFF: "{:.10g}".format,
-                Headers.GAS_HOLDUP_FRACTION_LOW_CUTOFF: "{:.10g}".format,
-                Headers.GAS_HOLDUP_FRACTION_HIGH_CUTOFF: "{:.10g}".format,
-                Headers.ALPHA_MAIN: "{:.10g}".format,
-                Headers.ALPHA_PILOT: "{:.10g}".format,
-            }
-        try:
-            output_string = df_temp.to_string(index=False, justify="justify", formatters=formatters, header=header)
-        except ValueError:
-            pass
+        formatters = {
+            Headers.STRENGTH: "{:.10g}".format,
+            Headers.SCALE_FACTOR: "{:.10g}".format,
+            Headers.ROUGHNESS: "{:.10g}".format,
+            Headers.CONNECTION_FACTOR: "{:.10g}".format,
+            "CONNECTION_FACTOR": "{:.10g}".format,
+            Headers.FORMATION_PERMEABILITY_THICKNESS: "{:.10g}".format,
+            "FORMATION_PERMEABILITY_THICKNESS": "{:.10g}".format,
+            Headers.MEASURED_DEPTH: "{:.3f}".format,
+            "MD": "{:.3f}".format,
+            Headers.TRUE_VERTICAL_DEPTH: "{:.3f}".format,
+            "TVD": "{:.3f}".format,
+            Headers.START_MEASURED_DEPTH: "{:.3f}".format,
+            "START_MD": "{:.3f}".format,
+            Headers.END_MEASURED_DEPTH: "{:.3f}".format,
+            "END_MD": "{:.3f}".format,
+            Headers.FLOW_COEFFICIENT: "{:.10g}".format,
+            "CV": "{:.10g}".format,
+            Headers.FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
+            "FLOW_CROSS_SECTIONAL_AREA": "{:.3e}".format,
+            Headers.OIL_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
+            Headers.GAS_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
+            Headers.WATER_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
+            Headers.MAX_FLOW_CROSS_SECTIONAL_AREA: "{:.3e}".format,
+            Headers.DEFAULTS: "{:.10s}".format,
+            Headers.WATER_HOLDUP_FRACTION_LOW_CUTOFF: "{:.10g}".format,
+            Headers.WATER_HOLDUP_FRACTION_HIGH_CUTOFF: "{:.10g}".format,
+            Headers.GAS_HOLDUP_FRACTION_LOW_CUTOFF: "{:.10g}".format,
+            Headers.GAS_HOLDUP_FRACTION_HIGH_CUTOFF: "{:.10g}".format,
+            Headers.ALPHA_MAIN: "{:.10g}".format,
+            Headers.ALPHA_PILOT: "{:.10g}".format,
+        }
+
+        if header:
+            # Modify headers to reduce width.
+            column_splits = [tuple(column.split("_")) for column in df_temp.columns]
+            number_of_levels = max([len(tup) for tup in column_splits])
+            if number_of_levels > 1:
+                formatters.update(
```

Comment:
I get a linter warning that mapping does not have update as a method, I think dict concatenation/merging does the same thing e.g. formatters = {**formatters, **{<new dict>}}.